src/label.cpp

```diff
@@ -17,27 +17,29 @@
 NAMESPACE_BEGIN(nanogui)
 
 Label::Label(Widget *parent, const std::string &caption, const std::string &font, int fontSize)
-    : Widget(parent), mCaption(caption), mFont(font) {
-    if (mTheme) {
-        mFontSize = mTheme->mStandardFontSize;
+    : Widget(parent, font, false), mCaption(caption) {
+
+    if (mTheme)
         mColor = mTheme->mTextColor;
-    }
-    if (fontSize >= 0) mFontSize = fontSize;
+
+    if (fontSize >= 0)
+        setFontSize(fontSize);
 }
 
 void Label::setTheme(Theme *theme) {
     Widget::setTheme(theme);
-    if (mTheme) {
-        mFontSize = mTheme->mStandardFontSize;
+
+    if (mTheme)
         mColor = mTheme->mTextColor;
-    }
 }
 
 Vector2i Label::preferredSize(NVGcontext *ctx) const {
     if (mCaption == "")
         return Vector2i::Zero();
-    nvgFontFace(ctx, mFont.c_str());
-    nvgFontSize(ctx, fontSize());
+
+    float fontSize = Widget::fontSize(mTheme->mStandardFontSize);
```

Comment:
First occurrence of many, `Widget::fontSize` has changed meaning.  Supply a default value (`mTheme->{mStandardFontSize,mTextBoxFontSize}` etc).  If `Widget::mFontSize` has been set explicitly via `setFontSize`, then `mFontSize` is returned.  Otherwise, the default value specified by the caller is returned.

All font sizes have been changed to floats, since the `nvgFontSize` method parameter is a `float` (e.g., `Theme` font sizes are all floats now but with same values).

Reply:
Sounds good!